src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js

```diff
@@ -76,7 +89,56 @@ export default class UserSettingsContainer extends Component {
       router.push(nextPage);
     }
     untouch('preferredName');
-  };
+  }
+
+  uploadPicture(pictureFile) {
+    return cashay.mutate('createUserPicturePutUrl', {
+      variables: {
+        userFilename: pictureFile.name
```

Comment:
Or `path.extname`! We could also just sent the MIME type (and require a mime type...). I'm kindof a fan of that idear. What do you think?